src/components/vr-scatter-plot/vr-scatter-plot.component.ts

```diff
@@ -1,32 +1,101 @@
+import { PreferenceService } from '../../services/preference/preference.service';
+import { DataService } from '../../services/data/data.service';
+import { Meta } from '../../datasets/metas/types';
+import { LineChartDatum} from '../line-chart/line-chart.component';
+import { LegendItemStyle as LineChartLegendItemStyle } from '../../d3/line-chart.d3';
+import { VRScatterplotMeta } from '../../datasets/metas/vr-scatter-plot.meta';
+import { LineChartMeta } from '../../datasets/metas/line-chart.meta';
+import { TimeSeriesDatum, TimeSeriesQueryOptions } from '../../datasets/queries/time-series.query';
+import { VRTimeSeriesDatum,VRTimeSeriesQueryOptions } from '../../datasets/queries/vr-time-series.query';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { DAY } from '../../utils/timeUnits';
+import { map, takeUntil } from 'rxjs/operators';
+import { DATA_PREFERENCE } from '../../i18n';
 import 'aframe';
+import { datasets } from '../../datasets';
 
+export type VRScatterplotDatum = VRTimeSeriesDatum<LineChartLegendItemStyle>;
 
 
 @Component({
   selector: 'app-vr-scatter-plot',
-  templateUrl: './vr-scatter-plot.component.html'
+  templateUrl: './vr-scatter-plot.component.html',
+  styleUrls: ['./vr-scatter-plot.component.scss'],
 })
-export class VRScatterPlotComponent implements OnInit, OnChanges, OnDestroy{
-  private vrScatterPlot: Scatterplot;
-  private shape: string;
-  private color: string;
-
+export class VRScatterPlotComponent<T extends Meta>implements OnInit, OnChanges, OnDestroy{
+  endDate = new Date();
+  startDate = new Date(this.endDate.getTime() - 30 * DAY);
+  datasetPref: VRScatterplotMeta | LineChartMeta;
+  dataset$ = this.preferenceService.dataset$;
+  DATA_PREFERENCE = DATA_PREFERENCE;
+  vrScatterPlot: Scatterplot;
+  componentMetas: Meta[];
+  exitVR: boolean = false;
+  queryOptions$ = new BehaviorSubject<VRTimeSeriesQueryOptions | TimeSeriesQueryOptions>({
+    range: [this.startDate, this.endDate],
+  });
+  datum$ = new BehaviorSubject<VRScatterplotDatum | LineChartDatum >({
+    label: '',
+    points: [],
+  });
+  private destroy$ = new Subject();
+ 
   constructor(
+    private preferenceService: PreferenceService
   ) {
-    this.vrScatterPlot = new Scatterplot('a-sphere');
+    this.preferenceService = preferenceService;
   }
+
   ngOnInit() {
-   this.vrScatterPlot.init(document.querySelector('a-scene'), [10, 20, 30, 40, 50, 60]);
+    this.vrScatterPlot = new Scatterplot('a-sphere');
+    const dataService = new DataService(this.preferenceService);
+    dataService.dataset$
+    .pipe(takeUntil(this.destroy$))
+    .subscribe(dataset => {
+      // componentMetas is initialized to different dataset metas - will help funnel dataset
+      this.componentMetas = dataset.metas;
+      // dataset.metas[0].type = 'tabbed' and dataset.metas[1] = 'line' if chosen UserWhiteNoise
+      // dataset.metas[0] - 'line' if Dummy chosen
+
+      if ((dataset.metas[0] as Meta).type === 'tabbed') {
```

Comment:
I think you can make this entire block a `switch` statement.

Reply:
Now that the datasets have changed to be specific to either VR or 2D charts, what is the best way to handle types outside of `MetaType.SCATTER_PLOT` (both `TABBED_CHART`, `LINE_CHART`, `GEO_MAP`)?  I know this problem will also come up in @64json @jcc37 @HarrisonPearl projects.

Reply:
I'm thinking we can send in a blank dataset to our `d3` files and send an HTML dialog box that the user needs to choose a different dataset bc the current selection doesn't exist for that visualization type.

Either that, or we can even think of changing `PreferenceService` to extract information from the component (which seems kind of backward), where we can restrict the `options` field in the `name` part of `createPreference` to only include keys that pertain to our visualization (https://github.com/googleinterns/guide-doge/blob/master/src/services/preference/preference.service.ts#L25)